### PR TITLE
[EN-461] Fixes logos and email from addresses for OrderMailer with multiple stores

### PR DESCRIPTION
**Pull request by aplegatt**

### CODE REVIEW

**Review by kaliszKrzysztof — APPROVED** [review]

File: core/app/helpers/spree/mail_helper.rb [hunk 1]
```diff
@@ -11,11 +11,15 @@ def name_for(order)
...
+    def logo_store
+      @order&.store || current_store
+    end
+
     def logo_path
-      if current_store.present? && current_store.logo.attached? && current_store.logo.variable?
-        main_app.url_for(current_store.logo.variant(resize: '244x104>'))
-      elsif current_store.present? && current_store.logo.attached? && current_store.logo.image?
-        main_app.url_for(current_store.logo)
+      if logo_store.present? && logo_store.logo.attached? && logo_store.logo.variable?
```
Comment (przemosk) [inline core/app/helpers/spree/mail_helper.rb:19]: wdyt to add here guard clause ?

File: core/app/helpers/spree/mail_helper.rb [hunk 2]
```diff
@@ -11,14 +11,23 @@ def name_for(order)
       order.name || Spree.t('customer')
     end
 
+    def logo_store
+      @order&.store || current_store
```
Comment (przemosk) [inline core/app/helpers/spree/mail_helper.rb:15]: according to method name, should return `logo`

```suggestion
      @order&.store.logo || current_store.logo
```

Comment (aplegatt) [reply core/app/helpers/spree/mail_helper.rb:15]: This is supposed to return a store which logo should be used.

Comment (przemosk) [reply core/app/helpers/spree/mail_helper.rb:15]: in that case, name of this method is little misleading, could we change that ?

**Review by przemosk — APPROVED** [review]

**Review by nnande — APPROVED** [review]

**Review by damianlegawiec — APPROVED** [review]

**Pull request merged**